src/views/Checkout.vue

```diff
@@ -107,24 +108,33 @@ export default class Checkout extends Vue {
             const wallets = this.wallets.slice(0);
 
             // Generate a new array with references to the respective wallets' accounts
-            const accounts = wallets.reduce((accs, wallet) => {
-                accs.push(...wallet.accounts.values());
-                return accs;
-            }, [] as AccountInfo[]);
+            const accountsAndContracts = wallets.reduce((acc, wallet) => {
+                acc.push(...wallet.accounts.values());
+                acc.push(...wallet.contracts);
+                return acc;
+            }, [] as Array<AccountInfo | ContractInfo>);
 
             // Reduce userfriendly addresses from that
-            const addresses = accounts.map((account) => account.userFriendlyAddress);
+            const addresses = accountsAndContracts.map((accountOrContract) => accountOrContract.userFriendlyAddress);
 
             // Get balances through pico consensus, also triggers head-change event
             const network = (this.$refs.network as Network);
             const balances: Map<string, number> = await network.connectPico(addresses);
 
-            // Update accounts with their balances
+            // Update accounts/contracts with their balances
             // (The accounts are still references to themselves in the wallets' accounts maps)
-            for (const account of accounts) {
-                const balance = balances.get(account.userFriendlyAddress);
+            for (const accountOrContract of accountsAndContracts) {
+                const balance = balances.get(accountOrContract.userFriendlyAddress);
                 if (balance === undefined) continue;
-                account.balance = Nimiq.Policy.coinsToSatoshis(balance);
+
+                if ('type' in accountOrContract && accountOrContract.type === Nimiq.Account.Type.VESTING) {
+                    // Calculate available amount for vesting contract
+                    accountOrContract.balance = (accountOrContract as VestingContractInfo)
+                        .calculateAvailableAmount(this.height, Nimiq.Policy.coinsToSatoshis(balance));
+                    continue;
```

Comment:
I'd strongly prefer an else block here instead of continue, or event `? :` and `const isVesting = ....`